app/models/concerns/foreman_openscap/openscap_proxy_core_extensions.rb

```diff
@@ -4,18 +4,22 @@ module OpenscapProxyCoreExtensions
 
     included do
       validate :openscap_proxy_has_feature
+      validate :scap_client_class_present
+      after_save :update_scap_client
     end
 
-    def update_scap_client_params(proxy_id)
-      new_proxy = SmartProxy.find proxy_id
+    def update_scap_client
+      update_scap_client_params if openscap_proxy_id_changed? && openscap_proxy_id
```

Comment:
> If we want to add/remove it together with openscap_proxy, it will not be displayed correctly.

I don't think we want. Selecting proxy does not say if the host should or should not run foreman_scap_client, it only tells which proxy is used to proxy the communication. It only affects lookup values in my eyes.

> If you just delete lookup values for host, the Puppet will IMHO use defaults

yes, which in this case won't be probably set anywhere which means `server` and `port` values will be missing in ENC output. That cause failure during puppet run since [puppet manifest](https://github.com/theforeman/puppet-foreman_scap_client/blob/master/manifests/init.pp#L38) does not define any default values. User in this case will see failed report which I think is correct.

> If you still care about the host, you will remove the foreman_scap_client manually, or you will delete the host completely. 

I don't think user would manually associate or disassociate this puppet manifest, they do it in policy wizard through host group and they therefore they can't remove it in host form. They can't even in host group form.

> Scenario when someone removes the proxy and keeps the foreman_scap_client does not make much sense to me.

If there's an advanced user who knows which lookup values must be set, he or she can have custom logic e.g. defining per host group, environment etc. Therefore they might want to disassociate the proxy later, in this case we'd keep overriding it on host level if we didn't delete lookup values.

> Setting lookup values to something that will cause errors when disassociating proxy is probably a good idea, that way we can catch when someone disassociated proxy "by accident".

Which is the case when you delete them. Unless advanced user set them on different level, which again would be correct.